addons/website_sale/controllers/main.py

```diff
@@ -171,7 +171,10 @@ def get_attribute_value_ids(self, product):
     def _get_search_order(self, post):
         # OrderBy will be parsed in orm and so no direct sql injection
         # id is added to be sure that order is a unique sort key
-        return 'website_published desc,%s , id desc' % post.get('order', 'website_sequence desc')
+        order_value = post.get('order')
+        if not isinstance(order_value, str):
```

Comment:
This does not protect against non-existing fields, e.g. http://377533-11-0-24a8bc.runbot16.odoo.com/shop?order=pouet+asc&category=3

Reply:
True, so basically we should check this too...

Reply:
I just retested it with my patch and nope it is not needed as it is already done by the internal function that no ValueError is raised...